src/main/java/com/microsoft/sqlserver/jdbc/spark/DataFrameBulkRecord.java

```diff
@@ -37,7 +37,10 @@ public DataFrameBulkRecord(Iterator<Row> iterator, ColumnMetadata[] dfColumnMeta
              // Columns may be reordered between SQLTable and DataFrame. dfFieldIndex maps to the
              // corresponding column in rowData and thus use dfFieldIndex to get the column.
              int dfFieldIndex = dfColumnMetadata[idx].getDfColIndex();
-             rowData[idx] = row.get(dfFieldIndex);
+             boolean isAutoIncrement = dfColumnMetadata[idx].isAutoIncrement();
```

Comment:
can avoid the isAutoIncrement variable here.

Reply:
Done